# <font size='6' color='darkorange' face='Courier New'>**Demographic Data Analyzer**</font>


In [1]:
# импорт библиотеки pandas
import pandas as pd

In [2]:
# чтение файла с данными и сохранение в df
# получение первых 5 строк таблицы df
df = pd.read_csv('adult.data.csv')
df.head()

age         workclass  fnlwgt  education  education-num  \
0   39         State-gov   77516  Bachelors             13   
1   50  Self-emp-not-inc   83311  Bachelors             13   
2   38           Private  215646    HS-grad              9   
3   53           Private  234721       11th              7   
4   28           Private  338409  Bachelors             13   

       marital-status         occupation   relationship   race     sex  \
0       Never-married       Adm-clerical  Not-in-family  White    Male   
1  Married-civ-spouse    Exec-managerial        Husband  White    Male   
2            Divorced  Handlers-cleaners  Not-in-family  White    Male   
3  Married-civ-spouse  Handlers-cleaners        Husband  Black    Male   
4  Married-civ-spouse     Prof-specialty           Wife  Black  Female   

   capital-gain  capital-loss  hours-per-week native-country salary  
0          2174             0              40  United-States  <=50K  
1             0             0              13  United-States  <=50K  
2             0             0              40  United-States  <=50K  
3             0             0              40  United-States  <=50K  
4             0             0              40           Cuba  <=50K

## Сколько людей каждой расы представлено в этом наборе данных?  

Это должна быть серия Pandas с названиями рас в качестве индексных меток (расовая колонка)

In [3]:
race_quantity = df['race'].value_counts()
race_quantity

White                 27816
Black                  3124
Asian-Pac-Islander     1039
Amer-Indian-Eskimo      311
Other                   271
Name: race, dtype: int64

## Каков средний возраст мужчин?

In [4]:
# сначала посмотрим сколько общее количество мужчин и женщин
df['sex'].value_counts(dropna = False)

Male      21790
Female    10771
Name: sex, dtype: int64

In [5]:
# определяем средний возраст (до 2 знака после запятой)
middle_age_male = round(df[df['sex'] == 'Male']['age'].mean(), 2)
middle_age_male

39.43

## Каков процент людей, имеющих степень бакалавра?

In [6]:
pct_bachelors = round(df['education'].value_counts(normalize = True)['Bachelors'] * 100, 1)
pct_bachelors

16.4

## Какой процент людей с высшим образованием (бакалавров, магистров или докторантов) зарабатывает более 50 тыс.

In [9]:
# посмотрим содержание колонки
df['education'].unique()

array(['Bachelors', 'HS-grad', '11th', 'Masters', '9th', 'Some-college',
       'Assoc-acdm', 'Assoc-voc', '7th-8th', 'Doctorate', 'Prof-school',
       '5th-6th', '10th', '1st-4th', 'Preschool', '12th'], dtype=object)

In [8]:
# рассчитываем процент людей с в.о.
pct_top_education = round(df['education'].value_counts(normalize=True)
                  [['Bachelors', 'Masters', 'Doctorate']].sum() * 100, 1)
pct_top_education

23.0

In [10]:
# получим процент людей с в.о. зарабатывает > 50т
top_education_50 = round(df[df['education'].isin(['Bachelors', 'Masters', 'Doctorate'])]
                      ['salary'].value_counts(normalize = True)['>50K'] * 100, 1)
top_education_50

46.5

## Какой процент людей без высшего образования зарабатывают более 50 тыс?

In [11]:
low_education_luxurious = round(df[~df['education'].isin(['Bachelors', 'Masters', 'Doctorate'])]
                               ['salary'].value_counts(normalize = True)['>50K'] * 100, 1)
low_education_luxurious

17.4

## Какое минимальное количество часов человек работает в неделю?

In [13]:
minhours_job = df['hours-per-week'].min()
minhours_job

1

## Какой процент людей, которые работают минимальное количество часов в неделю, имеют зарплату более 50 тыс?

In [14]:
# определим процент людей которые работают с минимальным количеством часов
df[df['hours-per-week'] == 1].shape[0]

20

In [15]:
# теперь рассчитаем процент людей с з.п. более 50 тыс.
luxurious_pct = round(df[df['hours-per-week'] == 1]
                       ['salary'].value_counts(normalize = True)['>50K'] * 100, 1)
luxurious_pct

10.0

## В какой стране самый высокий процент людей, зарабатывающих более 50 тыс., и каков этот процент?

In [22]:
# смотрим сначала страну
luxuriousmax_country = (df[df['salary'] == '>50K']['native-country'].value_counts() /
                          df['native-country'].value_counts() * 100).sort_values(ascending = False)
luxuriousmax_country.index[0]

'Iran'

In [23]:
# теперь рассчитаем процент людей
luxuriousmax_country_pct = round(luxuriousmax_country.iloc[0], 1)
luxuriousmax_country_pct

41.9

## Определите самую популярную профессию среди тех, кто зарабатывает > 50 тыс. в Индии

In [24]:
favored_India_profession = df[(df['native-country'] == 'India') &
                          (df['salary'] == '>50K')]['occupation'].value_counts().index[0]
favored_India_profession

'Prof-specialty'

# ИТОГО:

- Количество людей каждой рассы

<table>
	<tr>
	    <th>Расса</th>
	    <th>Количество</th>
	</tr >
  	<tr>
	    <td >How many White </td>
	    <td>27816</td>
	</tr>
  	<tr>
	    <td >Black </td>
	    <td>3124</td>
	</tr>
  	<tr>
	    <td >Asian-Pac-Islander</td>
	    <td>1039</td>
	</tr>
  	<tr>
	    <td >Amer-Indian-Eskimo</td>
	    <td>311</td>
	</tr>
  	<tr>
	    <td >Other</td>
	    <td>271</td>
</table>

- Средний возраст мужчин - 39.43
- Процент людей, имеющих степень бакалавра - 16.4
- Процент людей с высшим образованием (бакалавров, магистров или докторантов) зарабатывает более 50 тыс. - 46.5
- Процент людей без высшего образования зарабатывают более 50 тыс. - 17.4
- Минимальное количество часов который работает специалист в неделю - 1
- Процент людей, которые работают минимальное количество часов в неделю, имея зарплату более 50 тыс. - 10.0
- В этой стране самый высокий процент людей, зарабатывающих более 50 тыс. - Iran 41.9
- Самая популярная профессия среди тех, кто зарабатывает более 50 тыс. в Индии  - Prof-specialty.


<hr style="margin-bottom: 40px;">